### PR TITLE
[BottomSheet] Add elevation property

### PR DESCRIPTION
**Pull request by codeman7**
Adds an elevation property to MDCBottomSheet and MDCBottomSheet presentation controller. Snapshot test have been added for the `MDCBottomSheetControlller.view` as well as unit test for all public APIs.

Prework for #8025

### CODE REVIEW

File: components/BottomSheet/src/MDCBottomSheetTransitionController.m [hunk 1]
```diff
@@ -194,4 +196,8 @@ - (UIAccessibilityTraits)scrimAccessibilityTraits {
   return _scrimAccessibilityTraits;
 }
 
+- (void)setElevation:(MDCShadowElevation)elevation {
```
Comment (romoore) [inline components/BottomSheet/src/MDCBottomSheetTransitionController.m]: Can this be removed? It looks like the default behavior for autosynthesized setters.

File: components/BottomSheet/tests/snapshot/MDCBottomSheetSnapshotTests.m [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+#import "MaterialBottomSheet.h"
+
+#import <UIKit/UIKit.h>
+#import <XCTest/XCTest.h>
+
+#import "MaterialSnapshot.h"
+
+/** Snapshot test for @c MDCBottomSheetController. */
+@interface MDCBottomSheetSnapshotTests : MDCSnapshotTestCase
```
Comment (romoore) [inline components/BottomSheet/tests/snapshot/MDCBottomSheetSnapshotTests.m]: Nit: Please name this `MDCBottomSheetControllerSnapshotTests` so it's clear it's a test for the controller.

**Review by romoore — COMMENTED** [review]
I'm not 100% confident, but shouldn't there be unit tests for whatever properties are passed-through on each of the classes?

File: components/BottomSheet/src/MDCBottomSheetTransitionController.m (hunk 1 above)
Comment (codeman7) [reply components/BottomSheet/src/MDCBottomSheetTransitionController.m]: Done

File: components/BottomSheet/tests/snapshot/MDCBottomSheetSnapshotTests.m (hunk 2 above)
Comment (codeman7) [reply components/BottomSheet/tests/snapshot/MDCBottomSheetSnapshotTests.m]: Done

**Comment by codeman7** [comment]
@romoore Test have been added for all pass through properties and for their respective APIs.

**Review by yarneo — CHANGES_REQUESTED** [review]
It looks like you are adding a shadow elevation to both the bottom sheet controller MDCShapedView (that has a shadow layer inside of it), and to MDCDraggableView. Wouldn't just one of them suffice; why both?

**Comment by codeman7** [comment]
@yarneo The MDCShapedShadow layer is used as the view of the MDCBottomSheetController. But, clients may use only the presentation controller and if they did then the would be using the MDCDraggableView as well.

**Review by yarneo — APPROVED** [review]

**Comment by yarneo** [comment]
Could be worth noting in the bottomSheetController elevation property not to set the presentationController elevation property but only the elevation property on the controller.

File: components/BottomSheet/src/MDCBottomSheetPresentationController.h [hunk 3]
```diff
@@ -14,6 +14,7 @@
 
 #import <UIKit/UIKit.h>
 #import "MDCBottomSheetController.h"
+#import "MaterialShadowElevations.h"
```
Comment (yarneo) [inline components/BottomSheet/src/MDCBottomSheetPresentationController.h]: can be removed

File: components/BottomSheet/src/MDCBottomSheetTransitionController.h [hunk 4]
```diff
@@ -58,6 +60,11 @@
  */
 @property(nonatomic, assign) CGFloat preferredSheetHeight;
 
+/**
+ The elevation of the bottom sheet. Defaults to @c MDCShadowElevationModalBottomSheet.
+ */
+@property(nonatomic, assign) MDCShadowElevation elevation;
```
Comment (yarneo) [inline components/BottomSheet/src/MDCBottomSheetTransitionController.h]: is this still needed?

File: components/BottomSheet/src/private/MDCDraggableView.m [hunk 5]
```diff
@@ -30,6 +31,10 @@ @interface MDCDraggableView () <UIGestureRecognizerDelegate>
 
 @implementation MDCDraggableView
 
++ (Class)layerClass {
```
Comment (yarneo) [inline components/BottomSheet/src/private/MDCDraggableView.m]: is this still needed?

Comment (codeman7) [reply components/BottomSheet/src/private/MDCDraggableView.m]: Removed.

File: components/BottomSheet/src/MDCBottomSheetTransitionController.h (hunk 4 above)
Comment (codeman7) [reply components/BottomSheet/src/MDCBottomSheetTransitionController.h]: Removed

File: components/BottomSheet/src/MDCBottomSheetPresentationController.h (hunk 3 above)
Comment (codeman7) [reply components/BottomSheet/src/MDCBottomSheetPresentationController.h]: Removed

**Review by yarneo — APPROVED** [review]
LGTM, just clean up clang comments

**Pull request merged**